src/bokeh/server/views/autoload_js_handler.py

```diff
@@ -54,11 +54,14 @@ class AutoloadJsHandler(SessionHandler):
     '''
 
     def set_default_headers(self):
-        self.set_header("Access-Control-Allow-Headers", "*")
-        self.set_header('Access-Control-Allow-Methods', 'PUT, GET, OPTIONS')
         self.set_header("Access-Control-Allow-Origin", "*")
+        self.set_header("Access-Control-Allow-Headers", "*")
+        self.set_header("Access-Control-Allow-Credentials", "true")
```

Comment:
Similar to `Access-Control-Allow-Headers`, this just indicates that we accept credentials. This doesn't require them to be present.

Reply:
Should this be configurable or conditioned on anything? Are there any reasons to not set this by default?

Reply:
From [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials):

> Credentials are cookies, [TLS](https://developer.mozilla.org/en-US/docs/Glossary/TLS) client certificates, or authentication headers containing a username and password. By default, these credentials are not sent in cross-origin requests, and **doing so can make a site vulnerable to [CSRF](https://developer.mozilla.org/en-US/docs/Glossary/CSRF) attacks.** (emphasis mine)

I don't think we can avoid this here. If we want the ability to pass credentials through, this option needs to be set. Having done so of course opens up the possibility of someone forging credentials. However, unless the application actually uses credentials for active sessions, this is not less secure than it was before.

Meaning, even by hardcoding the acceptance here, existing applications are as safe as before given that they couldn't have used credentials before anyway. For new applications that use this, general security guidelines apply.